### PR TITLE
Add default attachments for emails

### PR DESCRIPTION
**Pull request by tockata**

### CODE REVIEW

**Comment by viandrea** [comment]
@tockata moved this issue into todo.

File: client/Elyon.Fastly.EmailJob.RestClient/Api/IStorageApi.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    /// </summary>
+    [PublicAPI]
+    public interface IStorageApi
+    {
+        /// <summary>
+        /// Add file to the storage
+        /// </summary>
+        /// <param name="file">The file to be added to the storage</param>
+        [Post("/api/storage")]
+        Task AddFileAsync(FileSpecModel file);
```
Comment (ivelin-links) [inline client/Elyon.Fastly.EmailJob.RestClient/Api/IStorageApi.cs]: This method has to return the hash of the file

File: src/Elyon.Fastly.EmailJob.Domain/Services/IAttachmentsService.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Elyon.Fastly.EmailJob.Domain.Dtos;
+
+namespace Elyon.Fastly.EmailJob.Domain.Services
+{
+    public interface IAttachmentsService : IBaseService
+    {
+        Task AddFileAsync(string fileName, string content);
```
Comment (ivelin-links) [inline src/Elyon.Fastly.EmailJob.Domain/Services/IAttachmentsService.cs]: Return the hash of the file (not guid, as we would be working with hashes in this service)

File: src/Elyon.Fastly.EmailJob/Controllers/StorageController.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            _mapper = mapper;
+        }
+
+        [HttpPost]
+        public async Task AddFileAsync(FileSpecModel file)
+        {
+            if (file == null)
+                throw new ArgumentNullException(nameof(file));
+
+            var doesFileExist = await _storageService
```
Comment (ivelin-links) [inline src/Elyon.Fastly.EmailJob/Controllers/StorageController.cs]: Don't do the check here, but in the service. Controller methods are pure thin proxy to the services.

File: src/Elyon.Fastly.EmailJob.DomainServices/AttachmentsService.cs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        public async Task AddFileAsync(string fileName, string content)
+        {
+            if (string.IsNullOrWhiteSpace(fileName))
+                throw new ArgumentNullException(nameof(fileName));
+            if (string.IsNullOrWhiteSpace(content))
+                throw new ArgumentNullException(nameof(content));
+
+            var fileContent = Convert.FromBase64String(content);
+            var fileHash = _xxHashInstance.ComputeHash(fileContent).AsBase64String();
+            await _repository
```
Comment (ivelin-links) [inline src/Elyon.Fastly.EmailJob.DomainServices/AttachmentsService.cs]: check if the file exists before adding it to the repository (CheckIfFileExists)

File: src/Elyon.Fastly.EmailJob.DomainServices/AttachmentsService.cs [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                .AddAttachment(new InsertFileDto
+                { 
+                    FileName = fileName,
+                    Content = fileContent,
+                    OriginalXXHash = fileHash
+                })
+                .ConfigureAwait(false);
+        }
```
Comment (ivelin-links) [inline src/Elyon.Fastly.EmailJob.DomainServices/AttachmentsService.cs:78]: return the hash of the file

File: src/Elyon.Fastly.EmailJob/Controllers/StorageController.cs [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        private readonly IMapper _mapper;
+
+        public StorageController(IAttachmentsService storageService, IMapper mapper)
+        {
+            _storageService = storageService;
+            _mapper = mapper;
+        }
+
+        [HttpPost]
+        public async Task AddFileAsync(FileSpecModel file)
```
Comment (ivelin-links) [inline src/Elyon.Fastly.EmailJob/Controllers/StorageController.cs]: Return the hash

**Review by ivelin-links — COMMENTED** [review]
Commented

**Review by ivelin-links — APPROVED** [review]
Approved

**Pull request merged**